packages/javascript/bh-shared-ui/src/views/PrivilegeZones/Certification/CertificationTable.tsx

```diff
@@ -0,0 +1,286 @@
+
+    const domainMap = useMemo(() => {
+        const map = new Map<string, string>();
+        for (const domain of availableEnvironments) {
+            map.set(domain.id, domain.name);
+        }
+        return map;
+    }, [availableEnvironments]);
+
+    const tagMap = useMemo(() => {
+        const map = new Map<number, string>();
+        for (const tag of assetGroupTags) {
+            map.set(tag.id, tag.name);
+        }
+        return map;
+    }, [assetGroupTags]);
+
+    const certificationsData = data ?? { pages: [{ count: 0, data: { members: [] } }] };
+    const count = certificationsData.pages[0].count;
+    const certificationsItemsRaw =
+        data?.pages.flatMap(
+            (page: { data?: { members: AssetGroupTagCertificationRecord[] } }) => page.data?.members ?? []
+        ) ?? [];
+    const totalFetched = certificationsItemsRaw.length;
+
+    const certificationsItems = isSuccess
+        ? certificationsItemsRaw.map((item: AssetGroupTagCertificationRecord) => {
+              return {
+                  ...item,
+                  date: DateTime.fromISO(item.created_at).toFormat('yyyy-MM-dd'),
+                  domainName: domainMap.get(item.environment_id) ?? 'Unknown',
+                  zoneName: tagMap.get(item.asset_group_tag_id) ?? 'Unknown',
+              };
+          })
+        : [];
+
+    const fetchMoreOnBottomReached = useCallback(
+        (containerRefElement?: HTMLDivElement | null) => {
+            if (containerRefElement) {
+                const { scrollHeight, scrollTop, clientHeight } = containerRefElement;
+                //once the user has scrolled within 500px of the bottom of the table, fetch more data if we can
+                if (scrollHeight - scrollTop - clientHeight < 500 && !isFetching && totalFetched < count) {
+                    fetchNextPage();
+                }
+            }
+        },
+        [fetchNextPage, isFetching, totalFetched, count]
+    );
+
+    useEffect(() => {
+        fetchMoreOnBottomReached(scrollRef.current);
+    }, [fetchMoreOnBottomReached]);
+
+    useEffect(() => {
+        // clear selection whenever the dropdown filter changes
+        setSelectedRows([]);
+    }, [dropdownSelection, setSelectedRows]);
+
+    const allSelected = certificationsItems?.length > 0 && selectedRows.length === certificationsItems?.length;
+    const someSelected = selectedRows.length > 0 && !allSelected;
+
+    const toggleAll = (checked: boolean) => {
+        onRowSelect(null);
+
+        if (checked) {
+            setSelectedRows(certificationsItems.map((row) => row.id));
+        } else {
+            setSelectedRows([]);
+        }
+    };
+
+    const toggleRow = (checked: boolean, selectedId: number) => {
+        const newSelectedRows = checked
+            ? [...selectedRows, selectedId]
+            : selectedRows.filter((rowId: number) => rowId !== selectedId);
+        setSelectedRows(newSelectedRows);
```

Comment:
id recommend using the set state callback to make sure the state and ui are always in sync!
```suggestion
        setSelectedRows(prev => {
        const newSelectedRows =  checked
            ? [...prev, selectedId]
            : prev.filter((rowId: number) => rowId !== selectedId);
            
        if (newSelectedRows...) ...
            
        return newSelectedRows
        })
```